### ISSUE TITLE
The Linux WARN regarding AVIC on PC Engines

### ISSUE THREAD

**Issue by pietrushnic** [Dasharo/dasharo-issues#1138]
I have found out that AVIC is not supported on this HW. The CPUID 0x8000000A EDX indicates no support for AVIC. That means the IOMMU guest AVIC support should not be exposed at all. I have prepared a fix for it already by hiding the guest AVIC capability in IOMMU and disabling the feature. So far it works and no more WARNs is visible in dmesg.

_Originally posted by @miczyg1 in https://github.com/Dasharo/dasharo-issues/issues/1134#issuecomment-2475797045_

**Comment by miczyg1** [comment]
Fixed it here: https://github.com/Dasharo/coreboot/pull/592

**Comment by pietrushnic** [comment]
@miczyg1 huge kudos for that :heart: 

Do you have ideas on how I would write a test against that? If not for the upcoming release, then for the next one, I would like to have it to avoid regression. I can do this for Dasharo (coreboot+SeaBIOS), but I'm not sure if it would be useful for Dasharo (coreboot+UEFI). Is just checking for the warning in dmesg good enough? I will check if it is visible in v24.08.00.01 while working on the release.

**Comment by miczyg1** [comment]
I guess it would have to be something to check dmesg for: `[   23.736476] WARNING: CPU: 2 PID: 1 at drivers/iommu/amd/init.c:980 enable_iommus_vapic+0` . Of course ignoring the timestamp, CPU number, PID and location in the file. For example if a line contains both `WARNING: CPU` and `enable_iommus_vapic`, the test should fail.

**Comment by pietrushnic** [comment]
> However I am not sure if this is reproducible  with the old test environment (we use quite old OSes/kernels)

Based on this comment, this should be moved to the milestone introducing OSFV to the test cycle. Does it make sense?

**Comment by miczyg1** [comment]
> > However I am not sure if this is reproducible  with the old test environment (we use quite old OSes/kernels)
> 
> Based on this comment, this should be moved to the milestone introducing OSFV to the test cycle. Does it make sense?

Probably, but then we can only validate the Dasharo (coreboot+UEFI).

**Comment by pietrushnic** [comment]
No, we have to introduce Dasharo (coreboot+SeaBIOS) testing in OSFV. Some steps towards that direction were already made.
